# Notebook 01 — Análisis Exploratorio de Datos (EDA)
**Curso:** BD-143 Programación II — CUC 2026
**Dataset:** TMDB Movies 2020–2025


## Carga de datos
Se cargó el dataset TMDB 2020-2025 con 9,999 películas y 15 columnas.
Se identificaron nulos principalmente en backdrop_path (18.81%) y overview (2.78%).

In [1]:
import sys
import os
sys.path.insert(0, os.path.abspath("../src"))

from eda.cargador_datos import CargadorDatos
from eda.procesador_eda import ProcesadorEda

ruta = "../data/raw/tmdb_2020_to_2025.csv"
cargador = CargadorDatos(ruta)
df = cargador.cargar()

Filas cargadas 9999
Valores nulos por columna:
                   cantidad  porcentaje %
adult                     0          0.00
backdrop_path          1881         18.81
genre_ids                 0          0.00
id                        0          0.00
original_language         0          0.00
original_title            0          0.00
overview                278          2.78
popularity                0          0.00
poster_path             243          2.43
release_date              0          0.00
title                     0          0.00
video                     0          0.00
vote_average              0          0.00
vote_count                0          0.00


## Resultados de la carga
- El dataset tiene 9,999 películas y 15 columnas
- Las columnas con más nulos son backdrop_path con 18.81% y overview con 2.78%
- El resto de columnas están completas sin nulos

## Limpieza de datos
Se aplica la limpieza del dataset: eliminación de columnas innecesarias,
conversión de tipos, manejo de nulos y normalización de géneros.

In [2]:
procesador = ProcesadorEda(df)
df_limpio = procesador.limpieza_datos()

Limpieza de datos completada
Columnas actuales: ['genre_ids', 'id', 'original_language', 'original_title', 'overview', 'popularity', 'release_date', 'title', 'vote_average', 'vote_count', 'anio_estreno']
Dataset limpio guardado en ../../data/processed/tmdb_movies_clean.csv


## Resultados de la limpieza

**Columnas eliminadas:**
- `Unnamed: 0` — índice sin valor informativo
- `adult` — todos los valores eran False, sin variación
- `video` — todos los valores eran False, sin variación
- `backdrop_path` — ruta de imagen, no útil para análisis (18.81% nulos)
- `poster_path` — ruta de imagen, no útil para análisis (2.43% nulos)

**Conversión de tipos:**
- `release_date` — convertido de string a datetime y se extrajo `anio_estreno`
- `genre_ids` — convertido de string a lista real con nombres de géneros

**Manejo de nulos:**
- `overview` — 278 nulos rellenados con "Sin descripción" ya que no es relevante para el EDA

**Dataset final:**
- 11 columnas y 9,999 filas
- Guardado en data/processed/tmdb_movies_clean.csv

## Resumen descriptivo
Estadísticas descriptivas de las variables numéricas principales del dataset.

In [3]:
procesador.resumen_descriptivo()

Resumen de descriptivo:
        popularity  vote_average    vote_count
count  9999.000000   9999.000000   9999.000000
mean      5.814750      4.551764    221.900790
std      13.101449      2.968713    879.465244
min       0.192900      0.000000      0.000000
25%       3.673650      0.000000      0.000000
50%       4.320600      5.708000      6.000000
75%       5.650550      6.724000     61.000000
max     951.346000     10.000000  21265.000000


,popularity,vote_average,vote_count
count,9999.000000,9999.000000,9999.000000
mean,5.814750,4.551764,221.900790
std,13.101449,2.968713,879.465244
min,0.192900,0.000000,0.000000
25%,3.673650,0.000000,0.000000
50%,4.320600,5.708000,6.000000
75%,5.650550,6.724000,61.000000
max,951.346000,10.000000,21265.000000


## Resultados del resumen descriptivo

**Popularidad:**
- Promedio de 5.8 pero máximo de 951, indica outliers significativos
- El 75% de las películas tiene popularidad menor a 5.65

**Calificación promedio (vote_average):**
- El 25% tiene calificación 0, correspondiente a películas sin votos aún
- La mediana real es 5.7 entre las películas calificadas

**Número de votos (vote_count):**
- La mitad de las películas tiene 6 votos o menos
- Máximo de 21,265 votos, hay películas muy votadas y otras casi sin votos
- Para análisis de calificaciones se filtra vote_count >= 50 para mayor confiabilidad

## Matriz de correlación
Analizamos la relación entre las variables numéricas del dataset.

In [4]:
procesador.matriz_correlacion()

Matriz de correlacion:
              popularity  vote_average  vote_count  anio_estreno
popularity      1.000000      0.042163    0.074511      0.125821
vote_average    0.042163      1.000000    0.162026      0.104922
vote_count      0.074511      0.162026    1.000000     -0.055091
anio_estreno    0.125821      0.104922   -0.055091      1.000000


,popularity,vote_average,vote_count,anio_estreno
popularity,1.000000,0.042163,0.074511,0.125821
vote_average,0.042163,1.000000,0.162026,0.104922
vote_count,0.074511,0.162026,1.000000,-0.055091
anio_estreno,0.125821,0.104922,-0.055091,1.000000


## Resultados de la matriz de correlación

- **popularity y vote_average (0.04)** — correlación muy baja, una película popular no necesariamente es bien calificada
- **popularity y vote_count (0.07)** — correlación baja, ser popular no garantiza muchos votos
- **vote_average y vote_count (0.16)** — correlación baja-moderada, las películas mejor calificadas tienden a tener más votos
- **anio_estreno y vote_count (-0.05)** — correlación negativa leve, películas más recientes tienen menos votos porque llevan menos tiempo en la plataforma
- En general las variables tienen baja correlación entre sí, lo que indica que son independientes y cada una aporta información diferente

## Películas por año
Distribución de películas estrenadas por año en el dataset.

In [5]:
procesador.peliculas_por_anio()

Peliculas por año:
anio_estreno
2020    1567
2021    1874
2022    1895
2023    1626
2024    1601
2025    1436
Name: id, dtype: int64


anio_estreno
2020    1567
2021    1874
2022    1895
2023    1626
2024    1601
2025    1436
Name: id, dtype: int64

## Resultados películas por año

- 2022 fue el año con más producciones con 1,895 películas
- 2020 tuvo menos producciones posiblemente por el impacto del COVID-19
- 2025 tiene menos películas porque el dataset fue extraído antes de que terminara el año
- En general la producción se mantiene estable entre 1,400 y 1,900 películas por año

## Top 10 películas más populares
Las películas con mayor puntuación de popularidad en TMDB.

In [6]:
procesador.top_peliculas_populares()

Top 10 películas más populares:
                                 title  popularity
0                         Frankenstein    951.3460
7                             Playdate    388.3856
2                   Predator: Badlands    314.6851
4                               Code 3    251.8624
3                        Black Phone 2    242.3412
6                    War of the Worlds    204.4781
5                       End of Loyalty    186.6110
8                                Abyss    179.5906
10  Chainsaw Man - The Movie: Reze Arc    154.4678
9                            Baramulla    148.1601


,title,popularity
0,Frankenstein,951.3460
7,Playdate,388.3856
2,Predator: Badlands,314.6851
4,Code 3,251.8624
3,Black Phone 2,242.3412
6,War of the Worlds,204.4781
5,End of Loyalty,186.6110
8,Abyss,179.5906
10,Chainsaw Man - The Movie: Reze Arc,154.4678
9,Baramulla,148.1601


## Resultados top películas más populares

- Frankenstein lidera con 951 puntos, casi el doble que la segunda película
- La mayoría son franquicias conocidas o secuelas muy esperadas
- Hay una gran diferencia entre Frankenstein (951) y el resto, confirmando que es un outlier
- La popularidad en TMDB se basa en visitas, listas y actividad reciente en la plataforma

## Top 10 películas mejor calificadas
Películas con mayor calificación promedio, filtrando aquellas con al menos 50 votos para mayor confiabilidad.

In [9]:
procesador.top_peliculas_calificadas()

Top 10 películas mejor calificadas:
                                 title  vote_average  vote_count
1759                          Succubus         9.019         188
2778  Attack on Titan: THE LAST ATTACK         8.800         130
2763        Selena Gomez: My Mind & Me         8.641         553
131                       Young Hearts         8.600         160
6206                 The King of Kings         8.600         109
3805                       Thriller 40         8.528          88
6437       Squid Game: Making Season 2         8.462         434
1805                    Love Untangled         8.447         123
5508                 Impossible Things         8.436         416
5662                              Pawn         8.433          67


,title,vote_average,vote_count
1759,Succubus,9.019,188
2778,Attack on Titan: THE LAST ATTACK,8.800,130
2763,Selena Gomez: My Mind & Me,8.641,553
131,Young Hearts,8.600,160
6206,The King of Kings,8.600,109
3805,Thriller 40,8.528,88
6437,Squid Game: Making Season 2,8.462,434
1805,Love Untangled,8.447,123
5508,Impossible Things,8.436,416
5662,Pawn,8.433,67


## Resultados top películas mejor calificadas

- Succubus lidera con 9.0 de calificación y 188 votos
- Attack on Titan y Squid Game aparecen como contenido de anime/series muy bien recibido
- Se aplicó un filtro de mínimo 50 votos para evitar películas con pocas calificaciones que distorsionen los resultados
- Las calificaciones rondan entre 8.4 y 9.0, todas consideradas excelentes

## Top 10 géneros más comunes
Los géneros que más aparecen en el dataset.

In [11]:
procesador.generos_mas_comunes()

Top 10 géneros más comunes:
genre_ids
Drama              4100
Comedia            2420
Thriller           1962
Terror             1533
Acción             1318
Romance            1185
Crimen              817
Documental          794
Ciencia ficción     668
Fantasía            635
Name: count, dtype: int64


genre_ids
Drama              4100
Comedia            2420
Thriller           1962
Terror             1533
Acción             1318
Romance            1185
Crimen              817
Documental          794
Ciencia ficción     668
Fantasía            635
Name: count, dtype: int64

## Resultados géneros más comunes

- Drama es el género dominante con 4,100 películas, casi el doble que Comedia
- Los 5 géneros más producidos son Drama, Comedia, Thriller, Terror y Acción
- Una película puede tener varios géneros, por eso los números superan el total de películas
- Géneros como Western y Música son los menos representados en el dataset

## Top 10 idiomas más comunes
Los idiomas originales más frecuentes en el dataset.

In [12]:
procesador.idiomas_mas_comunes()

Top 10 idiomas más comunes:
original_language
en    5566
fr     545
ja     520
es     470
it     275
tl     245
de     222
ko     221
zh     184
hi     175
Name: count, dtype: int64


original_language
en    5566
fr     545
ja     520
es     470
it     275
tl     245
de     222
ko     221
zh     184
hi     175
Name: count, dtype: int64

## Resultados idiomas más comunes

- El inglés domina ampliamente con 5,566 películas, representando el 55% del dataset
- Francia y Japón ocupan el segundo y tercer lugar con 545 y 520 películas respectivamente
- El español aparece en cuarto lugar con 470 películas
- Hay 81 idiomas distintos en total, pero los 10 primeros concentran la mayoría de la producción